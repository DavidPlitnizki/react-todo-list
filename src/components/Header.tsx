import React from 'react';
import AddIcon from '@mui/icons-material/Add';
import LoginIcon from '@mui/icons-material/Login';
import { Avatar, Button, Typography } from '@mui/material';
import Fab from '@mui/material/Fab';
import { Box } from '@mui/system';
import { styled as MUIStyled } from '@mui/system';
import styled from 'styled-components';

import {colors} from '../styles/colors';
import ContainerCenter from '../ui/ContainerCenter';
import ContainerContent from '../ui/ContainerContent';

const FabWrapper = styled('div')`
    justify-self: start;
    transform: translateY(2rem);
`;

const OwnFab = MUIStyled(Fab)({
    '&:hover': {
        cursor: 'pointer',
        background: colors.orange,
      }
})

const StyledContainerCenter = MUIStyled(ContainerCenter)({
    boxShadow: '0px 10px 30px -7px #000000, 5px 5px 15px 5px rgb(0 0 0 / 0%)'
})

const StyledContainerContent = MUIStyled(ContainerContent)({
    display: 'grid',
    gridTemplateColumns: 'repeat(3, 1fr)',
    gridTemplateRows: '4rem',
    justifyItems: 'center'
})

const StyledUserInfoBox = MUIStyled(Box)({
    paddingInline: '1rem',
})

const StyledEndGrid = MUIStyled(Box)({
    justifySelf: 'end'
})


interface IProps {
    onOpenTaskModal: (active: boolean) => void;
}

const Header:React.FC<IProps> = ({onOpenTaskModal}) => {

    const onHandleClick = () => {
        onOpenTaskModal(true)
    }

    return (
    <Box className='header-component'>
        <StyledContainerCenter bgColor={colors.bg}>
            <StyledContainerContent className='container__content'>
                <FabWrapper>
                        <OwnFab color='primary' aria-label="add" onClick={onHandleClick}>
                            <AddIcon />
                        </OwnFab>
                </FabWrapper>
                <h1>TO DO LIST</h1>
                <StyledEndGrid>
                    <ContainerCenter padding={false}>
                        <Avatar sx={{ bgcolor:'#B8B42D' }}>OP</Avatar>
                        <StyledUserInfoBox>
                            <Typography variant='body2'>User Name</Typography>
                            <Button variant="contained" endIcon={<LoginIcon />}>
                                Login
                            </Button>
                        </StyledUserInfoBox>
                    </ContainerCenter>
                </StyledEndGrid>
            </StyledContainerContent>
        </StyledContainerCenter>
    </Box>
    )
}

export default Header;