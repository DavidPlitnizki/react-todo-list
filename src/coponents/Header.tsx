import React from 'react';
import ContainerSpaceBetween from '../ui/ContainerSpaceBetween';
import Container from '../ui/FlexContainer';
import ButtonComponent from '../ui/ButtonComponent';


interface IProps {
    onOpenTaskModal: (active: boolean) => void;
}

const Header:React.FC<IProps> = ({onOpenTaskModal}) => {
    return (
    <Container className='header-wrapper'>
        <ContainerSpaceBetween>
            <ButtonComponent text='Add Task' onHandle={onOpenTaskModal} />
            <div>search</div>
            <div>sort</div>
        </ContainerSpaceBetween>
    </Container>
    )
}

export default Header;