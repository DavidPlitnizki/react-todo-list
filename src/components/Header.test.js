import {render, screen } from "@testing-library/react";
import React from "react";

import Header from "./Header";

describe('should show title', () => {
    test('should show TO DO LIST', () => {
        const {asFragment} = render(<Header />);
        const text = screen.getByText(/TO DO LIST/i);
        expect(text).toBeInTheDocument();

        expect(asFragment(<Header />)).toMatchSnapshot();
    })
});