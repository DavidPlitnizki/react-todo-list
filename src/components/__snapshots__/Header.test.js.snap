// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show title should show TO DO LIST 1`] = `
<DocumentFragment>
  <div
    class="header-component MuiBox-root css-0"
  >
    <div
      class="sc-bcXHqe eQHqsd css-bjpy14"
    >
      <h1>
        TO DO LIST
      </h1>
      <div
        class="sc-gswNZR dMvYgs"
      >
        <button
          aria-label="add"
          class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary css-gnsr6b-MuiButtonBase-root-MuiFab-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
